pkg/statemachine/stateMachine.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package statemachine
+
+type StateDirection int64
+
+const (
+	Forward StateDirection = iota
+	Backward
+	Stop
+)
+
+type StateMachine struct {
+	index    int
+	states   []string
+	finished bool
+}
+
+func NewStateMachine(states []string) *StateMachine {
+	return &StateMachine{
+		states: states,
+	}
+}
+
+func (sm *StateMachine) CurrentState() string {
+	if sm.index < 0 || sm.index >= len(sm.states) {
+		return ""
+	}
+	return sm.states[sm.index]
+}
+
+func (sm *StateMachine) NextState(direction StateDirection) string {
+	switch direction {
+	case Forward:
+		sm.index++
+	case Backward:
+		sm.index--
+	default:
+		return ""
```

Comment:
...then here we could also return the same error string constant...

Reply:
avoided for NextState to return anything so the only function to do that is CurrentState

Reply:
?

`NextState` _is_ returning something. The implicit assumption that the empty string is an error isn't very intuitive and needs documentation. If you don't want to return an error, at least a string representing an error condition is much clearer to understand.

Reply:
Github was showing old code.